app/routes/administrative-units/administrative-unit/core-data/index.js

```diff
@@ -90,14 +91,29 @@ export default class AdministrativeUnitsAdministrativeUnitCoreDataIndexRoute ext
     }
 
     let kboContacts = A();
-
     let kboAdministrativeUnit = await administrativeUnit.kboAdministrativeUnit;
+
+    if (!kboAdministrativeUnit) {
+      const identifiers = await administrativeUnit.identifiers;
+      const kboIdentifier = identifiers.find((id) => {
+        return id.idName === ID_NAME.KBO;
+      });
+      const structuredIdKBO = await kboIdentifier.structuredIdentifier;
+      console.log(structuredIdKBO);
+      const getKboData = `/kbo-data-sync/${structuredIdKBO.id}/kbo`;
+      await fetch(getKboData, {
+        method: 'POST',
+      });
+      kboAdministrativeUnit = await administrativeUnit.kboAdministrativeUnit;
+    }
+
```

Comment:
**Question:** Will we be keeping this functionality to retrieve the KBO data on the fly once the service periodically retrieves the data via a cronjob? If yes, I would move this to the route to create new administrative units similar to case as for the OVO numbers. That way we only retrieve the KBO data on demand for new administrative units perform overnight updates for already existing units (I presume this is what the cronjob would do, did not yet check ik in detail).